woocommerce/woocommerce
# [CYS on Core] Ensure the user is directed to the in app marketplace whenever clicking on 'Browse all themes'

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Whenever the store owner clicks on "Browse all themes", they should be directed to: `/wp-admin/admin.php?page=wc-admin&tab=themes&path=%2Fextensions` on core while at the same time keeping the functionality on WooExpress unchanged.

⚠️ While here we are still relying on the pre-existing `isWooExpress()` for this purpose, the plan is to switch this logic later on to an approach that is environment agnostic, relying on hooks instead for achieving all environment-specific changes.



Closes #45284



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Initial Setup

1. Enable the WooCommerce Beta Tester plugin if you don't have it enabled yet
2. On your Dashboard, head over to Tools > WCA Test Helper > Features and enable the `customize-store`

#### Test on Core

1. Access your Dashboard and head over to WooCommerce > Home
2. Click on the "Start customizing" or the "Customize Your Store" button
3. Click on the "Browse all themes" button:

<img width="1630" alt="Screenshot 2024-03-04 at 15 54 48" src="https://github.com/woocommerce/woocommerce/assets/15730971/e9419ba3-472f-4c33-bd60-d236f752247f">

4. Make sure you are redirected to `/wp-admin/admin.php?page=wc-admin&tab=themes&path=%2Fextensions` as demonstrated in the video below:

https://github.com/woocommerce/woocommerce/assets/15730971/7d3d0770-5dda-4e7c-b2c1-26dded0c2525

#### Test on WooExpress

1. If you don't have a WooExpress install yet, create one by following these steps: pdibGW-2pG-p2
2. Access your Dashboard and head over to WooCommerce > Home
3 Click on the "Start customizing" or the "Customize Your Store" button
4. Click on the "Browse all themes" button: 

<img width="1998" alt="Screenshot 2024-03-04 at 17 07 34" src="https://github.com/woocommerce/woocommerce/assets/15730971/17943bab-cd9a-4f15-a1da-bb81fb735288">

5. Make sure you are redirected to `https://wordpress.com/themes/{your_site_name}` as demonstrated in the video below:

https://github.com/woocommerce/woocommerce/assets/15730971/0b12a853-8059-4a7b-aa4e-d6848a1566b2



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Ensure the user is directed to the in-app marketplace whenever clicking on 'Browse all themes' within the Customize Your Store flow.

#### Comment 

</details>
